src/style.rs

```diff
@@ -97,14 +99,44 @@ pub struct TabBarStyle {
     /// Color of th line separating the tab name area from the tab content area.
     /// By `Default` it's [`Color32::BLACK`].
     pub hline_color: Color32,
+
+    /// Whether tab titles expand to fill the width of their tab bars.
+    pub fill_tab_bar: bool,
 }
 
-/// Specifies the look and feel of individual tabs.
+/// Specifies the look and feel of an individual tab.
 #[derive(Clone, Debug)]
 pub struct TabStyle {
-    /// Inner margin of tab body. By `Default` it's `Margin::same(4.0)`
-    pub inner_margin: Margin,
+    /// Style of the tab when it is active.
+    pub active: TabInteractionStyle,
+
+    /// Style of the tab when it is inactive.
+    pub inactive: TabInteractionStyle,
+
+    /// Style of the tab when it is focused.
+    pub focused: TabInteractionStyle,
+
+    /// Style of the tab when it is hovered.
+    pub hovered: TabInteractionStyle,
+
+    /// Style for the tab body.
+    pub tab_body: TabBodyStyle,
+
+    /// If `true`, show the hline below the active tabs name.
+    /// If `false`, show the active tab as merged with the tab ui area.
+    /// By `Default` it's `false`.
+    pub hline_below_active_tab_name: bool,
 
+    /// The prefered width of the tab. The tab will allways bet atleast
+    /// wide enough to fully show the title.
+    ///
+    /// Overriden by [`TabBarStyle::fill_tab_bar`]
```

Comment:
Could this dub as minimum tab width if `fill_tab_bar` is set? It would apply in the case where there are too many tabs to fit in the available area, which turns the tab bar into a scrollable area.

Reply:
Cool. Would be great if you included it in this PR before we merge it.

Reply:
I added this. Wasnt a big change and works great!